packages/editor/src/browser/editor-decoration.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2017 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { injectable, inject } from "inversify";
+import { Range } from "./editor";
+import { EditorManager } from "./editor-manager";
+import URI from "@theia/core/lib/common/uri";
+
+export const EditorDecorationsService = Symbol('EditorDecorationsService');
+
+export interface EditorDecorationsService {
+    registerDecorationType(decorationType: DecorationType): void;
+    removeDecorationType(key: string): void;
+    getDecorationTypes(): DecorationType[];
+    setDecorations(uri: URI, key: string, decorations: DecorationOptions[]): void;
```

Comment:
it should return `Disposable` i assume to return decorations

Reply:
the current flow is: 1) scan the document and identify list of decorations 2) apply new state. I do not see why this needs an extra step to remove them first. in case of monaco, it should be as cheap as possible, maybe that's why they just apply new state, instead of removing first. 

who and how would they be removed otherwise? I mean, we can introduce this layer to return a disposable, which will set an empty array. but which client will do that? current merge conflict decoration updater will already have a new state.

Reply:
sounds good